backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportResultData.java

```diff
@@ -118,6 +120,23 @@ public File writeToTemporaryFile(@NonNull final String filenamePrefix)
 		}
 	}
 
+	public Path writeToDirectory(@NonNull final Path directory)
+	{
+		final String ext = MimeType.getExtensionByType(reportContentType);
+		final String fileName = "Report_" + System.currentTimeMillis() + ext;
+		final Path file = directory.resolve(fileName);
```

Comment:
what's the reason for ignoring reportFilename?

Reply:
when i tested on local i found that all the reportFilename is allways `report.pdf`, so one report overwrites the others and we end up we just one file (the last one)